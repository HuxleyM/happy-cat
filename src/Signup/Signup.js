import React,{useContext} from 'react';
import Styles from './Signup.module.css';
import Form from './Form/Form';
import Confirmation from './Confirmation/Confirmation'
import {UserContext} from '../Context/userContext'

function Signup(){
    const {user,setUser} = useContext(UserContext)
    console.log('heya', user)
    
    return (
        <div className={Styles.signUpWrapper}>
            {!user && <div>Loading</div>} 
            {user.completed ? <Confirmation/>: <Form/>}
        </div>
    )
}

export default Signup