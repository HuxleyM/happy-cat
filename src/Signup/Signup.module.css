.signUpWrapper{
    border:2pt solid #c4c4c4;
    width: 1000px;
    height: 500px;
    border-radius: 10px;
}