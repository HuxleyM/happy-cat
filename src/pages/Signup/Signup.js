import React, { useContext } from "react";
import Styles from "./Signup.module.css";
import Form from "./Form/Form";
import Completed from "./Completed/Completed";
import Loading from "./Loading/Loading";
import { UserContext } from "../../Context/userContext";
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect,
} from "react-router-dom";

// Not best practice - use router or reducer or even a customer hook

const selectComponent = (user, componentToRender, route) => {
  if (!user) return <Loading />;
  if (user.completed) return <Completed />;
  if (!user.completed) return <Form />;
};

function Signup() {
  const { user } = useContext(UserContext);
  const ComponentToDisplay = selectComponent(user);

  return (
    <div className={Styles.signUpWrapper}>
      <Router>
        <Switch>
          <Route path="/completed">
            {user.completed ? <Completed /> : <Redirect to="/" />}
          </Route>
          <Route path="/">
            <Form />
          </Route>
        </Switch>
      </Router>
    </div>
  );
}

export default Signup;
