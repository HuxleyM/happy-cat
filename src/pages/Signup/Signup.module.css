.signUpWrapper {
  border: 2pt solid #c4c4c4;
  width: 60vw;
  height: 500px;
  border-radius: 10px;
  padding: 60px 100px;
  margin: 18px;
}

@media only screen and (max-width: 700px) {
  .signUpWrapper {
    border: none;
    padding: 2vh 3vw;
  }
}
