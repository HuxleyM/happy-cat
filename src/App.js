import React, {useState, useMemo} from 'react';
import './App.css';
import Styles from'./App.module.css';
import { UserContext } from './Context/userContext';
import Signup from './Signup/Signup'

function App() {
    const [user, setUser] = useState({
        completed: false,
    })
    const providerValue = useMemo(() => ({user, setUser}),[user,setUser]);
  return (
    <div className="App">
        <UserContext.Provider value={providerValue}>
            <header className={Styles.mainHeader}>
                <h1>HappyCat.</h1>
            </header>
            <body className={Styles.mainBody}>
                <Signup/>
            </body>
        </UserContext.Provider>
    </div>
  );
}

export default App;
