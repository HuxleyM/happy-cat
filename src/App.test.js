import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import Signup from './Signup/Signup'


let wrapped = shallow(<App></App>);

describe('App', () => {
  it('should render the Signup Component correctly', () => {  
    expect(wrapped.containsMatchingElement(<Signup />)).toEqual(true); 
    expect(wrapped).toMatchSnapshot();
  });

});


