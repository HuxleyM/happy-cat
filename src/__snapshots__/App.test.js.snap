// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the Signup Component correctly 1`] = `
<div
  className="App"
>
  <header
    className="mainHeader"
  >
    <h1>
      HappyCat.
    </h1>
  </header>
  <Signup />
</div>
`;
